resources/arduino_files/infineon_board/infineon_board.ino

```diff
@@ -0,0 +1,362 @@
+const int IS_1 = A1; // (PROFET-1 A1 = 15)
+const int IS_2 = A2; // (PROFET-2 A2 = 16)
+
+// Channel select
+const int DSEL_0 = 2; // PROFET-0
+const int DSEL_1 = 6; // PROFET-1
+
+// Enable Sensing
+const int DEN_0 = A4; // PROFET-0 (A4 = 18)
+const int DEN_1 = 5;  // PROFET-1
+const int DEN_2 = 9;  // PROFET-2
+
+const int relayArray[] = {RELAY_0, RELAY_1, RELAY_2, RELAY_3, RELAY_4};
+
+const int NUM_DS18 = 3; // Number of DS18B20 Sensors
+
+uint8_t sensors_address[NUM_DS18][8];
+
+// Temperature chip I/O
+OneWire ds(DS18_PIN);
+DallasTemperature sensors(&ds);
+
+// Setup DHT22
+DHT dht(DHT_PIN, DHTTYPE);
+
+int led_value = LOW;
+
+void setup() {
+  Serial.begin(9600);
+  Serial.flush();
+
+  pinMode(LED_BUILTIN, OUTPUT);
+
+  sensors.begin();
+
+  // Setup sense pins
+  pinMode(IS_0, INPUT);
+  pinMode(IS_1, INPUT);
+  pinMode(IS_2, INPUT);  
+
+  // Setup diagnosis enable pins
+  pinMode(DEN_0, OUTPUT);
+  pinMode(DEN_1, OUTPUT);
+  pinMode(DEN_2, OUTPUT);
+  
+  // Setup relay pins
+  pinMode(RELAY_0, OUTPUT);
+  pinMode(RELAY_1, OUTPUT);
+  pinMode(RELAY_2, OUTPUT);
+  pinMode(RELAY_3, OUTPUT);
+  pinMode(RELAY_4, OUTPUT);
+  
+  dht.begin();
+  
+  //ENABLE DIAGNOSIS AND SELECT CHANNEL
+  digitalWrite(DEN_0, HIGH);  // DEN_0 goes HIGH so Diagnosis enabled for PROFET0
+  digitalWrite(DEN_1, HIGH);  // DEN_1 goes HIGH so Diagnosis enabled for PROFET1
+  digitalWrite(DEN_2, HIGH);  // DEN_2 goes HIGH so Diagnosis enabled for PROFET2
+  
+  digitalWrite(DSEL_0, LOW); // DSEL_0 LOW reads PROFET 0_0. DSEL_0 HIGH reades PROFET 0_1
+  digitalWrite(DSEL_1, LOW); // DSEL_1 LOW reads PROFET 1_0. DSEL_1 HIGH reades PROFET 1_1  
+}
+
+// CharBuffer stores characters and supports (minimal) parsing of
+// the buffered characters.
+template <uint8_t kBufferSize>
+class CharBuffer {
+  public:
+    CharBuffer() {
+      Reset();
+    }
+    void Reset() {
+      write_cursor_ = read_cursor_ = 0;
+    }
+    bool Append(char c) {
+      if (write_cursor_ < buf_ + kBufferSize) {
+        buf_[write_cursor_++] = c;
+        return true;
+      }
+      return false;
+    }
+    bool Empty() {
+      return read_cursor_ >= write_cursor_;
+    }
+    char Next() {
+      return buf_[read_cursor_++];
+    }
+    char Peek() {
+      return buf_[read_cursor_];
+    }
+    bool ParseInt(int* output) {
+      int& v = *output;
+      v = 0;
+      size_t len = 0;
+      while (!Empty() && isdigit(Peek())) {
+        char c = Next();
+        v = v * 10 + c - '0';
+        ++len;
+        if (len > 5) {
+          return false;
+        }
+      }
+      return len > 0;
+    }
+    bool MatchAndConsume(char c) {
+      if (Empty() || Peek() != c) {
+        return false;
+      }
+      Next();
+      return true;
+    }
+
+  private:
+    char buf_[kBufferSize];
+    uint8_t write_cursor_;
+    uint8_t read_cursor_;
+};
+
+// Accumulates a line, parses it and takes the requested action if it is valid.
+class SerialInputHandler {
```

Comment:
FWIW, I have a thought about making this more resilient to leading garbage: record the time when the first byte arrives for a line of input. If too much time passes before a newline is received (5 seconds?), the input is discarded.

Alternately, we switch to a mode where each line has start of record and end of record marks (byte sequences used only for that purpose), and a checksum. Then we can verify input, and report back whether we accepted the input (fields in the json data for num currently buffered bytes, num correctly processed records and num discarded bytes).

To go along with that we'd need a simple command-line tool for interacting with that protocol.